libraries/bot-applicationinsights/src/main/java/com/microsoft/bot/applicationinsights/BotTelemetryClientImpl.java

```diff
@@ -0,0 +1,242 @@
+
+        telemetryClient.trackDependency(telemetry);
+    }
+
+    @Override
+    public void trackAvailability(String name, OffsetDateTime timeStamp, java.time.Duration duration, String runLocation, boolean success, String message, Map<String, String> properties, Map<String, Double> metrics) {
+
+    }
+
+    @Override
+    public void trackDependency(String dependencyTypeName, String target, String dependencyName, String data, OffsetDateTime startTime, java.time.Duration duration, String resultCode, boolean success) {
+
+    }
+
+    /**
+     * Logs custom events with extensible named fields.
+     * @param eventName A name for the event.
+     * @param properties Named string values you can use to search and classify events.
+     * @param measurements Measurements associated with this event.
+     */
+
+    public void trackEvent(String eventName, Map<String, String> properties, Map<String, Double> measurements)
+    {
+        EventTelemetry telemetry = new EventTelemetry(eventName);
+        if (properties != null)
+        {
+            for(Map.Entry<String, String> p : properties.entrySet())
+            {
+                telemetry.getProperties().put(p.getKey(), p.getValue());
+            }
+        }
+
+        if (measurements != null)
+        {
+            for(Map.Entry<String, Double> p : measurements.entrySet())
+            {
+                telemetry.getMetrics().put(p.getKey(), p.getValue());
+            }
+        }
+
+        telemetryClient.trackEvent(telemetry);
+    }
+
+    /**
+     * Logs a system exception.
+     * @param exception The exception to log.
+     * @param properties Named string values you can use to search and classify events.
+     * @param measurements Measurements associated with this event.
+     */
+
+    public void trackException(Exception exception, Map<String, String> properties, Map<String, Double> measurements)
+    {
+        ExceptionTelemetry telemetry = new ExceptionTelemetry(exception);
+        if (properties != null)
+        {
+            for(Map.Entry<String, String> p : properties.entrySet())
+            {
+                telemetry.getProperties().put(p.getKey(), p.getValue());
+            }
+        }
+
+        if (measurements != null)
+        {
+            for(Map.Entry<String, Double> p : measurements.entrySet())
+            {
+                telemetry.getMetrics().put(p.getKey(), p.getValue());
+            }
+        }
+
+        telemetryClient.trackException(telemetry);
+    }
+
+    /**
+     * Send a trace message.
+     * @param message Message to display.
+     * @param severityLevel Trace severity level <see cref="Severity"/>.
+     * @param properties Named string values you can use to search and classify events.
+     */
+
+    public void trackTrace(String message, Severity severityLevel, Map<String, String> properties)
+    {
+        TraceTelemetry telemetry = new TraceTelemetry(message);
```

Comment:
we needed to set the [severity](https://github.com/microsoft/botbuilder-dotnet/blob/main/libraries/Microsoft.Bot.Builder.ApplicationInsights/BotTelemetryClient.cs#L157) to this?

Reply:
Not sure about this, since the constructor of TraceTelemetry only has the String message, but maybe we will need the severity in the future. So I'm not sure 🤔

Reply:
I fixed this